dbms/src/Storages/StorageDistributed.h

```diff
@@ -36,8 +36,17 @@ class StorageDistributed : public ext::shared_ptr_helper<StorageDistributed>, pu
     static StoragePtr createWithOwnCluster(
         const std::string & table_name_,
         const ColumnsDescription & columns_,
-        const String & remote_database_,      /// database on remote servers.
-        const String & remote_table_,         /// The name of the table on the remote servers.
+        const String & remote_database_,       /// database on remote servers.
+        const String & remote_table_,          /// The name of the table on the remote servers.
+        ClusterPtr owned_cluster_,
+        const Context & context_);
+    
+    static StoragePtr createWithOwnCluster(
+        const std::string & table_name_,
+        const ColumnsDescription & columns_,
+        const String & remote_database_,       /// database on remote servers.
```

Comment:
Why we need these two parameters: remote_database, remote_table, if we have remote_table_function_ptr?